reduce only loss with dp (#11594)
PyTorchLightning/pytorch-lightning: CHANGELOG.md

- Changed `MisconfigurationException` to `ModuleNotFoundError` when `rich` isn't available ([#11360](https://github.com/PyTorchLightning/pytorch-lightning/pull/11360))

> - Enable reducing only loss when using DP ([#11594](https://github.com/PyTorchLightning/pytorch-lightning/pull/11594))

Reviewer: Could be written like this for people relying on the reduction.
```suggestion
- Disabled reducing training_step output other than loss when using DP ([#11594](https://github.com/PyTorchLightning/pytorch-lightning/pull/11594))
```

> - Disabled reducing training_step output other than loss when using DP ([#11594](https://github.com/PyTorchLightning/pytorch-lightning/pull/11594))

Reviewer 2: ```suggestion
- When using DP (data-parallel), Lightning will no longer automatically reduce all tensors returned in training_step; it will only reduce the loss unless `training_step_end` is overridden ([#11594](https://github.com/PyTorchLightning/pytorch-lightning/pull/11594))
```

> 

Reviewer 2: ```suggestion
- When using DP (data-parallel), the `training_epoch_end` hook will no longer receive reduced outputs from `training_step` and instead get the full tensor of results from all GPUs.
```

Some implementations of training_epoch_end will break, because they may assume a particular shape of tensors. I would add this extra changelog entry to make it absolutely clear that these hooks are affected.
